Flaky-test: AdditionalServletsTest.testEmptyStringAsExtractionDirectory

### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.


### Example failure

https://github.com/apache/pulsar/actions/runs/3530823792/jobs/5923383709

### Exception stacktrace

```
2022-11-23T09:52:32.2426617Z [ERROR] Tests run: 2, Failures: 1, Errors: 0, Skipped: 1, Time elapsed: 0.418 s <<< FAILURE! - in org.apache.pulsar.broker.web.plugin.servlet.AdditionalServletsTest
2022-11-23T09:52:32.2427429Z [ERROR] testEmptyStringAsExtractionDirectory(org.apache.pulsar.broker.web.plugin.servlet.AdditionalServletsTest)  Time elapsed: 0.018 s  <<< FAILURE!
2022-11-23T09:52:32.2428297Z java.lang.NullPointerException: Cannot invoke "org.apache.pulsar.broker.web.plugin.servlet.AdditionalServletDefinitions.servlets()" because "definitions" is null
2022-11-23T09:52:32.2429066Z 	at org.apache.pulsar.broker.web.plugin.servlet.AdditionalServlets.load(AdditionalServlets.java:89)
2022-11-23T09:52:32.2429916Z 	at org.apache.pulsar.broker.web.plugin.servlet.AdditionalServletsTest.testEmptyStringAsExtractionDirectory(AdditionalServletsTest.java:64)
2022-11-23T09:52:32.2430669Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2022-11-23T09:52:32.2431238Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
2022-11-23T09:52:32.2431862Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-11-23T09:52:32.2432372Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
2022-11-23T09:52:32.2432886Z 	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
2022-11-23T09:52:32.2433496Z 	at org.testng.internal.invokers.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:47)
2022-11-23T09:52:32.2434054Z 	at org.testng.internal.invokers.InvokeMethodRunnable.call(InvokeMethodRunnable.java:76)
2022-11-23T09:52:32.2434590Z 	at org.testng.internal.invokers.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
2022-11-23T09:52:32.2435071Z 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-11-23T09:52:32.2435553Z 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2022-11-23T09:52:32.2436078Z 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2022-11-23T09:52:32.2436483Z 	at java.base/java.lang.Thread.run(Thread.java:833)
```

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

--- a/pulsar-broker-common/src/test/java/org/apache/pulsar/broker/web/plugin/servlet/AdditionalServletsTest.java
+++ b/pulsar-broker-common/src/test/java/org/apache/pulsar/broker/web/plugin/servlet/AdditionalServletsTest.java
@@ -52,10 +52,8 @@ public class AdditionalServletsTest {
         AdditionalServletWithClassLoader as1 = mock(AdditionalServletWithClassLoader.class);
         AdditionalServletWithClassLoader as2 = mock(AdditionalServletWithClassLoader.class);
 
-        String originalTmpDirectory = System.getProperty("java.io.tmpdir");
         try (MockedStatic<AdditionalServletUtils> utils = mockStatic(AdditionalServletUtils.class)) {
-            String tmpDirectory = "/my/tmp/directory";
-            System.setProperty("java.io.tmpdir", tmpDirectory);
+            String tmpDirectory =  System.getProperty("java.io.tmpdir");
             utils.when(() -> AdditionalServletUtils.searchForServlets(
                     "/additionalServletDirectory", tmpDirectory)).thenReturn(definitions);
             utils.when(() -> AdditionalServletUtils.load(asm1, tmpDirectory)).thenReturn(as1);
@@ -65,8 +63,6 @@ public class AdditionalServletsTest {
 
             Assert.assertEquals(servlets.getServlets().get("AS1"), as1);
             Assert.assertEquals(servlets.getServlets().get("AS2"), as2);
-        } finally {
-            System.setProperty("java.io.tmpdir", originalTmpDirectory);
         }
     }
 

